Dockerfile

```diff
@@ -11,7 +11,13 @@ RUN apt-get update \
     && pip install ${PIP_OPTIONS} -U setuptools
 
 WORKDIR /workspaces
-COPY . .
+
+# Create (dummy) files required to run setup.py
+RUN mkdir optuna \
+    && touch README.md \
+    && echo '__version__ = "0.0.0"' > optuna/version.py
```

Comment:
This hack may affect application developers of Optuna.
`pip` recognized Optuna version as `0.0.0`, and it will be updated with the following `requirements.txt` unexpectedly.

```console
$ python -c 'import optuna; print(optuna.__version__)'
2.0.0

$ pip freeze | grep optuna
# Editable install with no version control (optuna==0.0.0)

$ echo "optuna>1.0.0" > requirements.txt
$ pip install -r requirements.txt
```

`master` branch
```console
Requirement already satisfied: optuna>1.0.0 in /workspaces (from -r requirements.txt (line 1)) (2.0.0)
...
```

PR 1639
```
...
Building wheels for collected packages: optuna
  Building wheel for optuna (setup.py) ... done
  Created wheel for optuna: filename=optuna-2.0.0-py3-none-any.whl size=312963 sha256=970776f84fbedcf5ba738587c7c206dd19468cf4743f2a89bbde2f965b183f39
  Stored in directory: /root/.cache/pip/wheels/22/8b/08/d32553e8cd416e1974ae704d41102b5a691c9612ad982b7991
Successfully built optuna
Installing collected packages: optuna
  Attempting uninstall: optuna
    Found existing installation: optuna 0.0.0
    Uninstalling optuna-0.0.0:
      Successfully uninstalled optuna-0.0.0
Successfully installed optuna-2.0.0
```

Reply:
@toshihikoyanase Thanks for the investigation. Given these side effects, seems like I should close this issue 👍